lib/cli/parse.rs

```diff
@@ -530,15 +415,6 @@ pub(super) fn payment_executable_deploy_item(
             requires[payment_entry_point] requires_empty[],
         (payment_path) requires[] requires_empty[payment_entry_point, payment_version],
     );
-    if !payment_args_simple.is_empty() && !payment_args_complex.is_empty() {
```

Comment:
Maybe you could fetch changes in 1.6 regarding this check ? to see if args simple and json are not both provided ? check_no_conflicting_arg_types

https://github.com/casper-ecosystem/casper-client-rs/blob/dev/lib/cli/parse.rs#L430

Reply:
I've added a slightly modified version of this function to catch conflicts for the sdk, but to also reflect the removal of args_complex